ast/attribute.go

```diff
@@ -0,0 +1,36 @@
+package ast
+
+type attribute struct {
+	id      []byte
+	classes [][]byte
+	attrs   map[string][]byte
+}
+
+// AttributeFormatFunc is a function that takes an node and returns a byte slice in whatever format the renderer needs.
+type AttributeFormatFunc func(id []byte, classes [][]byte, attrs map[string][]byte) []byte
+
+// SetAttribute sets the attribute in n, if n is a container.
+func SetAttribute(n Node, id []byte, classes [][]byte, attrs map[string][]byte) {
```

Comment:
while I understand the desire to abstract things, things would be slightly simpler and faster if ast.Attribute was publicly exposed. We wouldn't need the duplicate definition in parser, no need to create a copy, no need for  SetAttribute and FormatAttribute. Less things is better.